### PR TITLE
3667: Add GET for integrations

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/IntegrationsControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        [SetUp]
+        public void Setup()
+        {
+            _mockMediator = new Mock<IMediator>();
+            _controller = new IntegrationsController(_mockMediator.Object);
+        }
+
+        [Test]
+        public async Task ShouldGetIntegrations()
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/IntegrationsControllerTests.cs]: Use a testcase instead. Do one where it is null and one where it is a string

File: tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/IntegrationsControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            objectResult.Value.Should().BeOfType<IntegrationsResult>();
+
+            var integrationsResult = objectResult.Value as IntegrationsResult;
+            integrationsResult.Should().NotBeNull();
+            integrationsResult.Url.Should().Be(url);
+
+            _mockMediator.Verify(m => m.Send(It.Is<GetIntegrationsBySolutionIdQuery>(r => r.Id == SolutionId), It.IsAny<CancellationToken>()), Times.Once);
+            _mockMediator.VerifyNoOtherCalls();
+        }
+    }
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/IntegrationsControllerTests.cs]: Should also do an if empty test (if Integrations is empty)

File: tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/NHSD.BuyingCatalogue.Solutions.API.UnitTests.csproj [hunk 3]
```diff
@@ -19,7 +19,7 @@
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.4.0" />
         <PackageReference Include="Moq" Version="4.13.1" />
         <PackageReference Include="nunit" Version="3.12.0" />
-        <PackageReference Include="NUnit3TestAdapter" Version="3.15.1"/>
+        <PackageReference Include="NUnit3TestAdapter" Version="3.15.1" />
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/NHSD.BuyingCatalogue.Solutions.API.UnitTests.csproj]: This change does not need to be in the changeset

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/GetIntegrationsBySolutionIdTests.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        [TestCase("         ")]
+        [TestCase("")]
+        [TestCase(null)]
+        public async Task ShouldGetIntegrationsUrl(string url)
+        {
+            _integrationsUrl = url;
+            var result = await _context.GetIntegrationsBySolutionIdHandler.Handle(_query, _token).ConfigureAwait(false);
+            result.Url.Should().Be(_integrationsUrl);
+        }
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/GetIntegrationsBySolutionIdTests.cs]: Should do an empty test

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/GetIntegrationsBySolutionIdTests.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        [Test]
+        public void ShouldThrowWhenSolutionNotPresent()
+        {
+            _solutionExists = false;
+            var exception = Assert.ThrowsAsync<NotFoundException>( () => _context.GetIntegrationsBySolutionIdHandler.Handle(_query, _token));
+
+            _context.MockSolutionRepository.Verify(r => r.CheckExists(_solutionId, _token), Times.Once);
+            _context.MockSolutionRepository.VerifyNoOtherCalls();
+            _context.MockSolutionDetailRepository.VerifyNoOtherCalls();
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/GetIntegrationsBySolutionIdTests.cs]: Extra line. Not needed

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs [hunk 6]
```diff
@@ -65,6 +66,8 @@ internal class TestContext
 
         public GetRoadMapBySolutionIdHandler GetRoadMapBySolutionIdHandler => (GetRoadMapBySolutionIdHandler)_scope.GetRoadMapByIdHandler;
 
+        public GetIntegrationsBySolutionIdHandler GetIntegrationsBySolutionIdHandler => (GetIntegrationsBySolutionIdHandler)_scope.GetIntegrationsByIdHandler;
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs]: Should rename to have solutionId beforehand

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs [hunk 7]
```diff
@@ -201,6 +204,8 @@ private class Scope
 
             public IRequestHandler<GetRoadMapBySolutionIdQuery, IRoadMap> GetRoadMapByIdHandler { get; }
 
+            public IRequestHandler<GetIntegrationsBySolutionIdQuery, IIntegrations> GetIntegrationsByIdHandler { get; }
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs]: Same here: GetIntegrationsBySolutionIdHandler

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs [hunk 8]
```diff
@@ -298,6 +303,7 @@ private class Scope
                 IRequestHandler<UpdateHybridHostingTypeCommand, ISimpleResult> updateHybridHostingTypeHandler,
                 IRequestHandler<UpdateRoadmapCommand, ISimpleResult> updateRoadmapHandler,
                 IRequestHandler<GetRoadMapBySolutionIdQuery, IRoadMap> getRoadMapByIdHandler,
+                IRequestHandler<GetIntegrationsBySolutionIdQuery, IIntegrations> getIntegrationsByIdHandler,
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs]: And here :)

File: tests/NHSD.BuyingCatalogue.Testing.Data/Entities/SolutionDetailEntity.cs [hunk 9]
```diff
@@ -23,6 +23,9 @@ public sealed class SolutionDetailEntity : EntityBase
 
         public string RoadMap { get; set; }
 
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Testing.Data/Entities/SolutionDetailEntity.cs]: Extra space not needed

File: src/NHSD.BuyingCatalogue.Solutions.API/IntegrationsResult.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
+using Newtonsoft.Json;
+
+namespace NHSD.BuyingCatalogue.Solutions.API
+{
+    public sealed class IntegrationsResult
```
[the remark this answers is not in the document]
Comment (kat1906) [reply src/NHSD.BuyingCatalogue.Solutions.API/IntegrationsResult.cs]: 👍

File: src/NHSD.BuyingCatalogue.Solutions.API/RoadMapResult.cs [hunk 11]
```diff
@@ -1,6 +1,6 @@
 using Newtonsoft.Json;
 
-namespace NHSD.BuyingCatalogue.Solutions.API.ViewModels
+namespace NHSD.BuyingCatalogue.Solutions.API
 {
     public sealed class RoadMapResult
```
[the remark this answers is not in the document]
Comment (kat1906) [reply src/NHSD.BuyingCatalogue.Solutions.API/RoadMapResult.cs]: 👍

File: src/NHSD.BuyingCatalogue.Solutions.Application/Queries/GetIntegrationsBySolutionId/GetIntegrationsBySolutionIdHandler.cs [hunk 12]
```diff
@@ -0,0 +1,33 @@
+
```
[the remark this answers is not in the document]
Comment (kat1906) [reply src/NHSD.BuyingCatalogue.Solutions.Application/Queries/GetIntegrationsBySolutionId/GetIntegrationsBySolutionIdHandler.cs]: 👍

File: src/NHSD.BuyingCatalogue.Solutions.Application/Queries/GetIntegrationsBySolutionId/IntegrationsDto.cs [hunk 13]
```diff
@@ -0,0 +1,9 @@
+using NHSD.BuyingCatalogue.Solutions.Contracts;
+
+namespace NHSD.BuyingCatalogue.Solutions.Application.Queries.GetIntegrationsBySolutionId
+{
+    internal class IntegrationsDto : IIntegrations
```
[the remark this answers is not in the document]
Comment (kat1906) [reply src/NHSD.BuyingCatalogue.Solutions.Application/Queries/GetIntegrationsBySolutionId/IntegrationsDto.cs]: 👍

File: tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Integrations/Supplier Get Solution Integrations.3664.feature [hunk 14]
```diff
@@ -0,0 +1,45 @@
+Feature:  Display Marketing Page Form Solution Integrations Section
```
[the remark this answers is not in the document]
Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Integrations/Supplier Get Solution Integrations.3664.feature]: good spot 👀

File: tests/NHSD.BuyingCatalogue.Testing.Data/Entities/SolutionDetailEntity.cs (hunk 9 above)
Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.Testing.Data/Entities/SolutionDetailEntity.cs]: 👍

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs (hunk 8 above)
Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs]: 👍

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs (hunk 7 above)
Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs]: 👍

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs (hunk 6 above)
Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs]: 👍

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/GetIntegrationsBySolutionIdTests.cs (hunk 5 above)
Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/GetIntegrationsBySolutionIdTests.cs]: Oh yeah 👍

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/GetIntegrationsBySolutionIdTests.cs (hunk 4 above)
Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/GetIntegrationsBySolutionIdTests.cs]: I have put this in roadmap tests too

File: tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/IntegrationsControllerTests.cs (hunk 1 above)
Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/IntegrationsControllerTests.cs]: also added for roadmap
